TST: test_jd_add_subtract_round_trip fails with scale = 'utc', jds = (0.0, 0.0), delta = 2.220446049250313e-16

@mhvk , maybe another edge case uncaught by #13462 ?

Example log: https://github.com/astropy/astropy/runs/7664838295?check_suite_focus=true

```
_______________________ test_jd_add_subtract_round_trip ________________________

    @given(scale=sampled_from(STANDARD_TIME_SCALES),
>          jds=unreasonable_jd(),
           delta=floats(-10000, 10000))

.../astropy/time/tests/test_precision.py:502: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

scale = 'utc', jds = (0.0, 0.0), delta = 2.220446049250313e-16

    @given(scale=sampled_from(STANDARD_TIME_SCALES),
           jds=unreasonable_jd(),
           delta=floats(-10000, 10000))
    @example(scale='utc',
             jds=(0.0, 2.2204460492503136e-13),
             delta=6.661338147750941e-13)
    ...
    @example(scale='utc', jds=(0.0, 5.787592627370942e-13), delta=0.0)
    @example(scale='utc', jds=(1.0, 0.25000000023283064), delta=-1.0)
    def test_jd_add_subtract_round_trip(scale, jds, delta):
        jd1, jd2 = jds
        if scale == 'utc' and (jd1+jd2 < 1
                               or jd1+jd2+delta < 1):
            # Near-zero UTC JDs degrade accuracy; not clear why,
            # but also not so relevant, so ignoring.
            thresh = 100*u.us
        else:
            thresh = 2*dt_tiny
        t = Time(jd1, jd2, scale=scale, format="jd")
        try:
            with quiet_erfa():
                t2 = t + delta*u.day
                if abs(delta) >= np.finfo(float).eps:
>                   assert t2 != t
E                   AssertionError: assert <Time object: scale='utc' format='jd' value=0.0> != <Time object: scale='utc' format='jd' value=0.0>
E                   Falsifying example: test_jd_add_subtract_round_trip(
E                       scale='utc', jds=(0.0, 0.0), delta=2.220446049250313e-16,
E                   )
E                   
E                   You can reproduce this example by temporarily adding @reproduce_failure('6.54.1', b'AXicY2NmwAUOG0BoAAjDAP0=') as a decorator on your test case
E                   Highest target scores:
E                         1.3411e-05  (label=' (b-a).to_value(u.s), from TimeDelta')
E                        1.44293e-05  (label=' (a-b).to_value(u.s), from TimeDelta')

.../astropy/time/tests/test_precision.py:526: AssertionError
```

--- a/astropy/time/tests/test_precision.py
+++ b/astropy/time/tests/test_precision.py
@@ -509,21 +509,23 @@ def test_leap_stretch_mjd(d, f):
          delta=7.327471962526035e-12)
 @example(scale='utc', jds=(0.0, 5.787592627370942e-13), delta=0.0)
 @example(scale='utc', jds=(1.0, 0.25000000023283064), delta=-1.0)
-@example(scale='utc', jds=(0.0, 0.0), delta=2.220446049250313e-16)
+@example(scale='utc', jds=(0.0, 0.0), delta=2*2.220446049250313e-16)
 def test_jd_add_subtract_round_trip(scale, jds, delta):
     jd1, jd2 = jds
     if scale == 'utc' and (jd1+jd2 < 1
                            or jd1+jd2+delta < 1):
         # Near-zero UTC JDs degrade accuracy; not clear why,
         # but also not so relevant, so ignoring.
-        thresh = 100*u.us
+        minimum_for_change = 1e-9
+        thresh = minimum_for_change * u.day
     else:
+        minimum_for_change = np.finfo(float).eps
         thresh = 2*dt_tiny
     t = Time(jd1, jd2, scale=scale, format="jd")
     try:
         with quiet_erfa():
             t2 = t + delta*u.day
-            if abs(delta) >= np.finfo(float).eps:
+            if abs(delta) >= minimum_for_change:
                 assert t2 != t
             t3 = t2 - delta*u.day
             assert_almost_equal(t3, t, atol=thresh, rtol=0)

--- a/astropy/time/tests/test_precision.py
+++ b/astropy/time/tests/test_precision.py
@@ -509,6 +509,7 @@ def test_leap_stretch_mjd(d, f):
          delta=7.327471962526035e-12)
 @example(scale='utc', jds=(0.0, 5.787592627370942e-13), delta=0.0)
 @example(scale='utc', jds=(1.0, 0.25000000023283064), delta=-1.0)
+@example(scale='utc', jds=(0.0, 0.0), delta=2.220446049250313e-16)
 def test_jd_add_subtract_round_trip(scale, jds, delta):
     jd1, jd2 = jds
     if scale == 'utc' and (jd1+jd2 < 1

